### ISSUE TITLE
Lots of `verify` test failures in v3.12.0 merge

### ISSUE THREAD

**Issue by syoyo** [lighttransport/embree-aarch64#39]
OS: Jetson AGX(aarch64 linux)
Build config: `scripts/bootstrap-aarch64-linux.sh`

With Intel's `v3.12.0` merge, lots of `verify` tests fails.
 
It looks something is changed from v3.11.0 to v3.12.0

```
Embree Ray Tracing Kernels 3.12.0 (b2e495336782d9ff20e59bf1f45cc7633bda7c5f)
  Compiler  : GCC 7.5.0
  Build     : Release 
  Platform  : Linux (32bit)
  CPU       : Unknown CPU ( MRA MRA MRA)
   Threads  : 8
   ISA      : SSE SSE2 NEON 
   Targets  : 
   MXCSR    : FTZ=0, DAZ=0
  Config
    Threads : default
    ISA     : SSE SSE2 NEON 
    Targets :  (supported)
              SSE2  (compile time enabled)
    Features: intersection_filter 
    Tasking : internal_tasking_system 

================================================================================
  WARNING: "Flush to Zero" or "Denormals are Zero" mode not enabled 
           in the MXCSR control and status register. This can have a severe 
           performance impact. Please enable these modes for each application 
           thread the following way:

           #include "xmmintrin.h"
           #include "pmmintrin.h"

           _MM_SET_FLUSH_ZERO_MODE(_MM_FLUSH_ZERO_ON);
           _MM_SET_DENORMALS_ZERO_MODE(_MM_DENORMALS_ZERO_ON);
================================================================================


 [PASSED]
                                                       fast_allocator_regression_test ... [PASSED]
                                                         motion_derivative_regression ... [PASSED]
                                                            collision_regression_test ... [PASSED]
                                                                cache_regression_test ... [PASSED]
                                                         parallel_for_regression_test ... [PASSED]
                                                      parallel_reduce_regression_test ... [PASSED]
                                                       parallel_prefix_sum_regression ... [PASSED]
                                                     parallel_for_for_regression_test ... [PASSED]
                                          parallel_for_for_prefix_sum_regression_test ... [PASSED]
                                                   parallel_partition_regression_test ... [PASSED]
                                                           RadixSortRegressionTestU32 ... [PASSED]
                                                           RadixSortRegressionTestU64 ... [PASSED]
                                                         parallel_set_regression_test ... [PASSED]
                                                         parallel_map_regression_test ... [PASSED]
                                                           parallel_filter_regression ... [PASSED]
                                                          barrier_sys_regression_test ... [PASSED]
                                                                NEON.multiple_devices ... [FAILED]
                                                                      NEON.types_test ... [PASSED]
                                                                      NEON.get_bounds ...---------------- [FAILED]
                                                               NEON.get_linear_bounds ...---------------- [FAILED]
                                                                   NEON.get_user_data ... [FAILED]
                                                                   NEON.buffer_stride ...-------- [FAILED]
                                                                     NEON.empty_scene ...---------- [FAILED]
                                                                  NEON.empty_geometry ...---------- [FAILED]
                                                                           NEON.build ...---------- [FAILED]
                                                          NEON.overlapping_primitives ...---------- [FAILED]
                                                             NEON.new_delete_geometry ...----- [FAILED]
                                                                NEON.user_geometry_id ...----- [FAILED]
                                                         NEON.enable_disable_geometry ...----- [FAILED]
                                                                          NEON.update ...---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [FAILED]
                                                              NEON.build_garbage_geom ... [FAILED]
                                                           NEON.interpolate.triangles ...------ [FAILED]
                                                                NEON.interpolate.grid ...------ [FAILED]
                                                              NEON.interpolate.subdiv ...------ [FAILED]
                                                                NEON.interpolate.hair ...------ [FAILED]
                                                                    NEON.triangle_hit ...---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [FAILED]
                                                                        NEON.quad_hit ...---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [FAILED]
                                                             NEON.intersection_filter ...-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [FAILED]
                                                                      NEON.instancing ...-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [FAILED]
                                                                   NEON.inactive_rays ...---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [FAILED]
                                                            NEON.watertight_triangles ...------------------------------------------------------------------------------------------------ [FAILED]
                                                         NEON.watertight_triangles_mb ...------------------------------------------------------------------------------------------------ [FAILED]
                                                                NEON.watertight_quads ...------------------------------------------------------------------------------------------------ [FAILED]
                                                             NEON.watertight_quads_mb ...------------------------------------------------------------------------------------------------ [FAILED]
                                                                NEON.watertight_grids ...------------------------------------------------------------------------------------------------ [FAILED]
                                                             NEON.watertight_grids_mb ...------------------------------------------------------------------------------------------------ [FAILED]
                                                               NEON.watertight_subdiv ...!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! [PASSED]
                                                              NEON.ray_alignment_test ...------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [FAILED]
                                                                     NEON.point_query ...----------------------- [FAILED]
                                                               NEON.regression_static ... [FAILED]
                                                              NEON.regression_dynamic ... [FAILED]
                                                    NEON.regression_static_build_join ... [FAILED]
                                                   NEON.regression_dynamic_build_join ... [FAILED]
                                                NEON.regression_static_memory_monitor ... [FAILED]
                                               NEON.regression_dynamic_memory_monitor ... [FAILED]
                                                            NEON.geometry_state_tests ... [FAILED]
                                                   NEON.scene_modified_geometry_tests ... [FAILED]
                                                   NEON.sphere_filter_multi_hit_tests ... [PASSED]

                                                                         Tests passed: 19
                                                                         Tests failed: 5051
                                                             Tests failed and ignored: 96
```

Situation is also same when using this PR: https://github.com/lighttransport/embree-aarch64/pull/35

**Comment by syoyo** [comment]
FYI, x64 build(MaxISA: SSE2 + Internal Tasking system) passes

```
$CMAKE_BIN \
  -DCMAKE_BUILD_TYPE=RelWithDebInfo \
  -DCMAKE_INSTALL_PREFIX=$HOME/local/embree3 \
  -DCMAKE_C_COMPILER=clang \
  -DCMAKE_CXX_COMPILER=clang++ \
  -DEMBREE_ISPC_SUPPORT=Off \
  -DEMBREE_TASKING_SYSTEM=Internal \
  -DEMBREE_TUTORIALS=On \
  -DEMBREE_MAX_ISA=SSE2 \
  -DEMBREE_RAY_PACKETS=ON \
  -Bbuild -H.
```

```
Embree Ray Tracing Kernels 3.12.0 (b62a7b6e53c26217a15b59522ddd8abac4ddca3d)
  Compiler  : CLANG 9.0.0 (tags/RELEASE_900/final)
  Build     : Release 
  Platform  : Linux (64bit)
  CPU       : Unknown CPU (AuthenticAMD)
   Threads  : 32
   ISA      : XMM YMM SSE SSE2 SSE3 SSSE3 SSE4.1 SSE4.2 POPCNT AVX F16C RDRAND AVX2 FMA3 LZCNT BMI1 BMI2 
   Targets  : SSE SSE2 SSE3 SSSE3 SSE4.1 SSE4.2 AVX AVXI AVX2 
   MXCSR    : FTZ=1, DAZ=1
  Config
    Threads : default
    ISA     : XMM YMM SSE SSE2 SSE3 SSSE3 SSE4.1 SSE4.2 POPCNT AVX F16C RDRAND AVX2 FMA3 LZCNT BMI1 BMI2 
    Targets : SSE SSE2 SSE3 SSSE3 SSE4.1 SSE4.2 AVX AVXI AVX2  (supported)
              SSE2  (compile time enabled)
    Features: intersection_filter 
    Tasking : internal_tasking_system 

 [PASSED]
                                                       fast_allocator_regression_test ... [PASSED]
                                                         motion_derivative_regression ... [PASSED]
                                                            collision_regression_test ... [PASSED]
                                                                cache_regression_test ... [PASSED]
                                                         parallel_for_regression_test ... [PASSED]
                                                      parallel_reduce_regression_test ... [PASSED]
                                                       parallel_prefix_sum_regression ... [PASSED]
                                                     parallel_for_for_regression_test ... [PASSED]
                                          parallel_for_for_prefix_sum_regression_test ... [PASSED]
                                                   parallel_partition_regression_test ... [PASSED]
                                                           RadixSortRegressionTestU32 ... [PASSED]
                                                           RadixSortRegressionTestU64 ... [PASSED]
                                                         parallel_set_regression_test ... [PASSED]
                                                         parallel_map_regression_test ... [PASSED]
                                                           parallel_filter_regression ... [PASSED]
                                                          barrier_sys_regression_test ... [PASSED]
                                                                SSE2.multiple_devices ... [PASSED]
                                                                      SSE2.types_test ... [PASSED]
                                                                      SSE2.get_bounds ...++++++ [PASSED]
                                                               SSE2.get_linear_bounds ...++++++ [PASSED]
                                                                   SSE2.get_user_data ... [PASSED]
                                                                   SSE2.buffer_stride ...++++++++ [PASSED]
                                                                     SSE2.empty_scene ...++++++++++ [PASSED]
                                                                  SSE2.empty_geometry ...++++++++++ [PASSED]
                                                                           SSE2.build ...++++++++++ [PASSED]
                                                          SSE2.overlapping_primitives ...++++++++++ [PASSED]
                                                             SSE2.new_delete_geometry ...............................................................................................+................................+++.+ [PASSED]
                                                                SSE2.user_geometry_id ...+++++ [PASSED]
                                                         SSE2.enable_disable_geometry ...+++++ [PASSED]
                                                                          SSE2.update ...++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ [PASSED]
                                                              SSE2.build_garbage_geom ..................................................... [PASSED]
                                                           SSE2.interpolate.triangles ...++++++ [PASSED]
                                                                SSE2.interpolate.grid ...++++++ [PASSED]
                                                              SSE2.interpolate.subdiv ...++++++ [PASSED]
                                                                SSE2.interpolate.hair ...++++++ [PASSED]
                                                                    SSE2.triangle_hit ...++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ [PASSED]
                                                                        SSE2.quad_hit ...++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ [PASSED]
                                                             SSE2.intersection_filter ...++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ [PASSED]
                                                                      SSE2.instancing ...++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ [PASSED]
                                                                   SSE2.inactive_rays ...++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ [PASSED]
                                                            SSE2.watertight_triangles ...++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ [PASSED]
                                                         SSE2.watertight_triangles_mb ...++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ [PASSED]
                                                                SSE2.watertight_quads ...++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ [PASSED]
                                                             SSE2.watertight_quads_mb ...++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ [PASSED]
                                                                SSE2.watertight_grids ...++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ [PASSED]
                                                             SSE2.watertight_grids_mb ...++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ [PASSED]
                                                               SSE2.watertight_subdiv ...!!!!!!!!!!!!!!!!!++++++++++++++++++++++!!++!!!!++!!!!!!!!!!++++!+!!!!!!!!!!!!!!+++++++++++++++++ [PASSED]
                                                              SSE2.ray_alignment_test ...++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++ [PASSED]
                                                                     SSE2.point_query ...+++++++++++++++++++++++ [PASSED]
                                                               SSE2.regression_static ................................. [PASSED]
                                                              SSE2.regression_dynamic ................................. [PASSED]
                                                    SSE2.regression_static_build_join .................................. [PASSED]
                                                   SSE2.regression_dynamic_build_join .................................... [PASSED]
                                                SSE2.regression_static_memory_monitor ............................................................... [PASSED]
                                               SSE2.regression_dynamic_memory_monitor ............................................................... [PASSED]
                                                            SSE2.geometry_state_tests ... [PASSED]
                                                   SSE2.scene_modified_geometry_tests ... [PASSED]
                                                   SSE2.sphere_filter_multi_hit_tests ... [PASSED]

                                                                         Tests passed: 5118
                                                                         Tests failed: 0
                                                             Tests failed and ignored: 48
```

**Comment by syoyo** [comment]
The reason was missing NEON isa detection when creating Embree device with `"isa=neon"` configuration. Fixed in recent `aarch64-v3.12.0` release
